### PR TITLE
Update ObjectModel.php

### PR DESCRIPTION
**Pull request by wdammak**
Fix PHP7.4

### CODE REVIEW

File: classes/ObjectModel.php [hunk 1]
```diff
@@ -682,7 +682,7 @@ public function update($null_values = false)
                 if ($shop_exists) {
                     if (Shop::isFeatureActive() && Shop::getContext() != Shop::CONTEXT_SHOP) {
                         foreach ($fields as $key => $val) {
-                            if (!array_key_exists($key, (array)$this->update_fields)) {
+                             if (!property_exists((array)$this->update_fields, $key)) {
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: $this->update_fields is an array, cannot use property_exists

File: classes/ObjectModel.php [hunk 2]
```diff
@@ -1221,10 +1221,10 @@ public function getWebserviceParameters($ws_params_attribute_name = null)
 
         if (isset($this->{$ws_params_attribute_name}['associations'])) {
             foreach ($this->{$ws_params_attribute_name}['associations'] as $assoc_name => &$association) {
-                if (!array_key_exists('setter', $association) || (isset($association['setter']) && !$association['setter'])) {
+                if (!property_exists($association, 'setter') || (isset($association['setter']) && !$association['setter'])) {
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: $association is an array, cannot use property_exists

File: classes/ObjectModel.php [hunk 3]
```diff
@@ -1221,10 +1221,10 @@ public function getWebserviceParameters($ws_params_attribute_name = null)
...
                     $association['setter'] = Tools::toCamelCase('set_ws_'.$assoc_name);
                 }
-                if (!array_key_exists('getter', $association)) {
+                if (!property_exists($association, 'getter')) {
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: should not use property_exists

File: classes/ObjectModel.php [hunk 4]
```diff
@@ -1258,7 +1258,7 @@ public function getWebserviceParameters($ws_params_attribute_name = null)
             }
             if (isset($details['validate'])) {
                 $current_field['validateMethod'] = (
-                    array_key_exists('validateMethod', $resource_parameters['fields'][$field_name]) ?
+                    property_exists($resource_parameters['fields'][$field_name], 'validateMethod') ?
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: $resource_parameters['fields'][$field_name] is also an array, cannot use property_exists

File: classes/ObjectModel.php [hunk 5]
```diff
@@ -1797,7 +1797,7 @@ public static function hydrateCollection($class, array $datas, $id_lang = null)
         $rows = array();
         if ($datas) {
             $definition = ObjectModel::getDefinition($class);
-            if (!array_key_exists($definition['primary'], $datas[0])) {
+            if (!property_exists($datas[0], $definition['primary'])) {
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: $datas is List of data (multi-dimensional array), cannot use property_exists

**Review by shreesh-webkul — CHANGES_REQUESTED** [review]

File: classes/ObjectModel.php (hunk 3 above)
Comment (wdammak) [reply classes/ObjectModel.php]: done

File: classes/ObjectModel.php [hunk 6]
```diff
@@ -1104,7 +1104,7 @@ public static function displayFieldName($field, $class = __CLASS__, $htmlentitie
         }
 
         $key = $class.'_'.md5($field);
-        return ((is_array($_FIELDS) && array_key_exists($key, $_FIELDS)) ? ($htmlentities ? htmlentities($_FIELDS[$key], ENT_QUOTES, 'utf-8') : $_FIELDS[$key]) : $field);
+        return ((is_array($_FIELDS) && property_exists($_FIELDS, $key)) ? ($htmlentities ? htmlentities($_FIELDS[$key], ENT_QUOTES, 'utf-8') : $_FIELDS[$key]) : $field);
```
Comment (shreesh-webkul) [inline classes/ObjectModel.php]: @wdammak 
I forgot to mark one change. $_FIELDS is an array. cannot use property_exists.
Please remove this change also.

Thanks.

Comment (wdammak) [reply classes/ObjectModel.php]: done

**Review by shreesh-webkul — APPROVED** [review]
changes look good.

**Pull request merged**